Generate serializers for class LayoutRect, ScreenDataOverrides, MediaCapabilitiesInfo, MediaStreamRequest

arunsundarapple opened the pull request:
#### 3de90cce1c0256a072a3d55ffceca43bbc412fda
<pre>
Generate serializers for class LayoutRect, ScreenDataOverrides, MediaCapabilitiesInfo, MediaStreamRequest
<a href="https://bugs.webkit.org/show_bug.cgi?id=247648">https://bugs.webkit.org/show_bug.cgi?id=247648</a>
rdar://101308191

Reviewed by Alex Christensen.

* Source/WebCore/platform/MediaCapabilitiesInfo.h:
(WebCore::MediaCapabilitiesInfo::encode const): Deleted.
(WebCore::MediaCapabilitiesInfo::decode): Deleted.
* Source/WebCore/platform/graphics/LayoutRect.h:
(WebCore::LayoutRect::encode const): Deleted.
(WebCore::LayoutRect::decode): Deleted.
* Source/WebCore/platform/graphics/VP9Utilities.h:
(WebCore::ScreenDataOverrides::encode const): Deleted.
(WebCore::ScreenDataOverrides::decode): Deleted.
* Source/WebCore/platform/mediastream/MediaStreamRequest.h:
(WebCore::MediaStreamRequest::encode const): Deleted.
(WebCore::MediaStreamRequest::decode): Deleted.
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:

Canonical link: <a href="https://commits.webkit.org/256545@main">https://commits.webkit.org/256545@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/8ef3c1a362844737dbf7d5ffe86619072f444043

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/96095 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/5344 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/29144 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/105646 "Built successfully") | [  ~~🛠 🧪 win~~](https://ews-build.webkit.org/#/builders/10/builds/165976 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/100073 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/5464 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/34107 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/88474 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/101464 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/101756 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/4045 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/82701 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/31073 "Passed tests") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/85886 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/87798 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/73906 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/39836 "Built successfully") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/19324 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/37511 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/20673 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/4528 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/42107 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/60/builds/43272 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/43997 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/39935 "Passed tests") | | 
<!--EWS-Status-Bubble-End-->

achristensen07 commented on Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:
```diff
@@ -1327,3 +1327,50 @@ header: <WebCore/TextChecking.h>
+    WebCore::MediaStreamRequest::Type type;
+};
+#endif
+struct WebCore::MediaCapabilitiesDecodingInfo {
```
This needs to serialize the contents of the parent class.
struct WebCore::MediaCapabilitiesDecodingInfo : WebCore::MediaCapabilitiesInfo

arunsundarapple replied:
Done.

achristensen07 commented on Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:
```diff
@@ -1327,3 +1327,50 @@ header: <WebCore/TextChecking.h>
+    WebCore::MediaDecodingConfiguration supportedConfiguration;
+}
+
+struct WebCore::MediaCapabilitiesEncodingInfo {
```
Ditto

arunsundarapple replied:
Done.

arunsundarapple committed:
Generate serializers for class LayoutRect, ScreenDataOverrides, MediaCapabilitiesInfo, MediaStreamRequest
https://bugs.webkit.org/show_bug.cgi?id=247648
rdar://101308191

Reviewed by Alex Christensen.

* Source/WebCore/platform/MediaCapabilitiesInfo.h:
(WebCore::MediaCapabilitiesInfo::encode const): Deleted.
(WebCore::MediaCapabilitiesInfo::decode): Deleted.
* Source/WebCore/platform/graphics/LayoutRect.h:
(WebCore::LayoutRect::encode const): Deleted.
(WebCore::LayoutRect::decode): Deleted.
* Source/WebCore/platform/graphics/VP9Utilities.h:
(WebCore::ScreenDataOverrides::encode const): Deleted.
(WebCore::ScreenDataOverrides::decode): Deleted.
* Source/WebCore/platform/mediastream/MediaStreamRequest.h:
(WebCore::MediaStreamRequest::encode const): Deleted.
(WebCore::MediaStreamRequest::decode): Deleted.
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:

Canonical link: https://commits.webkit.org/256545@main
